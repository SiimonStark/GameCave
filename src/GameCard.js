import React, { Component } from 'react';

export default class GameCard extends Component {
  constructor(props) {
    super(props);
    this.state= {
      gameTitle: this.props.title,
      gameScore: this.props.metaScore,
      gameGenres: this.props.genres,
      gamePlatforms: this.props.platforms,
      gameMulti: this.props.multiplayer,
      gameImage: this.props.img,
      inLibrary: false
     }
  }
  toggleInfo = () => {
    this.props.setPopup(
      true,
      this.state.gameTitle,
      this.state.gameScore,
      this.state.gameGenres,
      this.state.gamePlatforms,
      this.state.gameMulti,
      this.state.gameImage
    );
  }
  changeLibraryStatus = () => {
    this.setState({ inLibrary: this.state.inLibrary ? false : true });
  }
  render() {
    let libraryButtonClass = "add-to-library " + this.state.inLibrary;
    let inLibraryStatus = " + ";
    inLibraryStatus = this.state.inLibrary ? " - " : " + ";
    const style = { backgroundImage: 'url(' + this.props.img + ')', backgroundSize: 'cover' };
    return(
      <div className="game-card">
        <section className="img" onClick={this.toggleInfo}>
          <div className="gamecover" style={style}></div>
        </section>
        <section className="text-container" onClick={this.toggleInfo}>
          <h4>{this.props.title}</h4>
        </section>
        <section className="btn">
          <button onClick={this.changeLibraryStatus} className={libraryButtonClass}>{inLibraryStatus}</button>
        </section>
      </div>
    )
  }
}