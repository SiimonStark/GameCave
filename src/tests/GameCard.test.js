import React from 'react';
import ReactDOM from 'react-dom';
import GameCard from '../GameCard';
import { shallow } from 'enzyme';

const game = {
  "title": "Dark Souls",
  "platforms": ["PC", "Playstation 4", "Xbox One"],
  "genres": ["Action", "Adventure", "RPG", "Fantasy"],
  "multiplayer": true,
  "metaScore": 64,
  "img": "https://upload.wikimedia.org/wikipedia/en/thumb/8/8d/Dark_Souls_Cover_Art.jpg/220px-Dark_Souls_Cover_Art.jpg"}

const setPopupMock = jest.fn();

describe('GameCard', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <GameCard {...game}
      setPopup={setPopupMock}/>
    );
  });
  it('should match snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it ('should have default state', () => {
    expect(wrapper.state()).toEqual({ 
      gameTitle: game.title,
      gameScore: game.metaScore,
      gameGenres: game.genres,
      gamePlatforms: game.platforms,
      gameMulti: game.multiplayer,
      gameImage: game.img,
      inLibrary: false
    });
  });

  it('should invoke setPopup() on click', () => {
    wrapper.find('.game-more').simulate('click');
    expect(setPopupMock).toBeCalled();
  });

  it('should update inLibrary status when button is clicked', () => {
    expect(wrapper.state('inLibrary')).toEqual(false);
    wrapper.find('.add-to-library').simulate('click');
    expect(wrapper.state('inLibrary')).toEqual(true);
  })
})