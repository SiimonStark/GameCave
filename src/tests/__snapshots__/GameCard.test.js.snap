// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCard should match snapshop 1`] = `
<div
  className="game-card"
>
  <section
    className="img"
  >
    <div
      className="gamecover"
      style={
        Object {
          "backgroundImage": "url(https://upload.wikimedia.org/wikipedia/en/thumb/8/8d/Dark_Souls_Cover_Art.jpg/220px-Dark_Souls_Cover_Art.jpg)",
          "backgroundSize": "cover",
        }
      }
    />
  </section>
  <section
    className="text-container"
  >
    <h4>
      Dark Souls
    </h4>
    <p
      className="game-more"
      onClick={[Function]}
    >
      More Info..
    </p>
  </section>
  <section
    className="btn"
  >
    <button
      className="add-to-library false"
      onClick={[Function]}
    >
       + 
    </button>
  </section>
</div>
`;
